src/modules/deal-guarantee/dto/create-deal-guarantee-request.dto.ts

```diff
@@ -0,0 +1,98 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { PROPERTIES } from '@ukef/constants';
+import { DateOnlyString } from '@ukef/helpers/date-only-string.type';
+import { IsISO8601, IsNotEmpty, IsOptional, Length, Matches, Min, MinLength } from 'class-validator';
+
+export type CreateDealGuaranteeRequest = CreateDealGuaranteeRequestItem[];
+
+export class CreateDealGuaranteeRequestItem {
+  @ApiProperty({
+    description: 'The identifier of the deal to create the guarantee for.',
+    example: '00000001',
+    minLength: 1,
+    maxLength: 10,
+  })
+  @Length(1, 10)
+  // TODO APIM-73: Should we remove dealIdentifier from the request body?
+  readonly dealIdentifier: string;
+
+  // TODO APIM-73: Is it okay that I have removed portfolioIdentifier from the request body?
+
+  @ApiProperty({
+    description: `The date that this guarantee will take effect. This will be replaced by today's date if a date in the past is provided.`,
+    type: Date,
+    format: 'date',
+  })
+  @Matches(/^\d{4}-\d{2}-\d{2}$/)
+  @IsISO8601({ strict: true })
+  readonly effectiveDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'An ACBS party identifier.',
+    minLength: 1,
+    maxLength: 10,
+    example: '00000002',
+  })
+  @Length(1, 10)
+  // TODO APIM-73: ACBS says it can be at most 8 - which is correct?
```

Comment:
ACBS's documentation says the max length for `limitKey` is 8 but the Mulesoft RAML says it is 10 - which should we use?

Reply:
Limit, party or obligor are pointing to Party object. Id is always 8 character long. Min 8 Max 8, so @Length(8)